pkg/controller/versions/strategy_v2_5.go

```diff
@@ -239,6 +243,23 @@ func (v *versionStrategyV2_5) validateExtensionProviders(spec *v2.ControlPlaneSp
 	return allErrors
 }
 
+func (v *versionStrategyV2_5) validateExtensionProviderZipkin(spec *v2.ControlPlaneSpec, allErrors []error) []error {
+	if spec.MeshConfig == nil || spec.MeshConfig.ExtensionProviders == nil {
+		return allErrors
+	}
+	// we don't allow to set both existing Jaeger/Stackdriver and extensionProvider.zipkin at the same time
+	if spec.Addons.Jaeger != nil || spec.Addons.Stackdriver != nil {
```

Comment:
This condition will not work as you expect, because `spec *v2.ControlPlaneSpec` comes from `status.appliedSpec`, which has Jaeger configured by default:
```
status:
  appliedSpec:
    addons:
      jaeger:
        install:
          ingress:
            enabled: true
          storage:
            type: Memory
        name: jaeger
```
So I suggest to check `spec.tracing.type`:
```suggestion
	if spec.IsJaegerEnabled() || spec.IsStackdriverEnabled() {
```